### ISSUE TITLE
Signatures with FileType 0 don't match on original HTML documents.

### ISSUE THREAD

**Issue by LSchuepbach** [Cisco-Talos/clamav#843]
Describe the bug
----------------

Trying to create a signature that would hit on a comment inside an HTML comment, I realized it was never hitting, regardless which FileType I was giving.

I wish that signatures with FileType 0 (any) would also match on the HTML regardless of normalization.

How to reproduce the problem
----------------------------
Using this `perdu.html` file
```
﻿<!DOCTYPE html>
<!-- saved from url=(0048)https://perdu.com/ -->
<html><head><title>Vous Etes Perdu ?</title></head><body><h1>Perdu sur l'Internet ?</h1><h2>Pas de panique, on va vous aider</h2><strong><pre>    * <----- vous &ecirc;tes ici</pre></strong></body></html>
```
And this signature `cherche.ndb` that should trigger on the "saved from url":
```
SAVEDFROMURL:0:*:73617665642066726f6d2075726c
```
It doesn't hit.

I am aware that HTML is normalized, but the comment doesn't appear in any of the normalized html files staying when using `---leave-temps`.  This is why I use FileType 0. I would expect that given it would still try to find an exact match on the unmodified HTML.

Here's the logs with [debug.log](https://github.com/Cisco-Talos/clamav/files/10785835/debug.log)

These comment are not dangerous themselves, but can be used to recognize some phishing toolkits.

**Comment by Sanesecurity** [comment]
There's a way of getting it to match but involves the use of an ftm file.

test.ndb:

Sanesecurity.text.savedfromurl:0:*:3c212d2d2073617665642066726f6d2075726

test.ftm:

0:0:3c212d2d2073617665642066726f6d2075726c:Mail:CL_TYPE_ANY:CL_TYPE_MAIL

clamscan --database=test.ftm --database=test.ndb test.file

Now matches as the ftm file tells the scanner to effectively scan the file as all.

Messy yes but I'm sure there is a better way...

**Comment by micahsnyder** [comment]
> I wish that signatures with FileType 0 (any) would also match on the HTML regardless of normalization.

I agree with you.  I feel it should match on the original (not normalized) HTML.  Of course, it may then bother people if it matches on both versions if using `--allmatch`.  I'm not sure what to do about that.

**Comment by LSchuepbach** [comment]
I have plenty of signatures with FileType 2 that matches multiples times because the files are extracted in different ways. I don't see it as a problem and haven't heard yet complains on that, but you have more feedback that me.
